flowzone.yml

```diff
@@ -218,7 +218,20 @@
     name: Checkout versioned commit
     uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
     with:
-      fetch-depth: ${{ needs.versioned_source.outputs.depth || 0 }}
+      fetch-depth: ${{ inputs.checkout_fetch_depth }}
+      # Note that fetch-tags is not currently working as described:
+      # https://github.com/actions/checkout/issues/1781
+      fetch-tags: true
+      submodules: "recursive"
+      # fallback to an invalid ref if the checkout ref is undefined
+      ref: "${{ needs.versioned_source.outputs.sha || '¯\_(ツ)_/¯' }}"
+      token: ${{ steps.gh_app_token.outputs.token || secrets.FLOWZONE_TOKEN }}
+
+  - &checkoutVersionedShaDeep # https://github.com/actions/checkout
```

Comment:
If we are adding this and using it where we know it's required, why do we still need the input?

Reply:
Because some balenaOS device repos require a full fetch depth for every job with submodules, not just the release note jobs.

So we expose the input, make the default 1, except where we know we need it to be 0 and ignore the input.

Reply:
Repos needing full fetch depth:
- https://github.com/balena-os/balena-trex/actions/runs/10019894679/job/27749653700?pr=184
- https://github.com/balena-os/balena-tgw/commit/56e0bceeee6dd7a0582bd29a8b4e9fab6842e059